sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -354,6 +354,58 @@ case class Pow(left: Expression, right: Expression)
   }
 }
 
+/**
+ * Performs the inverse operation of HEX.
+ * Resulting characters are returned as a byte array.
+ */
+case class UnHex(child: Expression) extends UnaryExpression with Serializable {
```

Comment:
this should just implement ExpectsInputTypes, and use BinaryType as the only type.

There will be a rule in https://github.com/apache/spark/pull/7175 to do implicit type casting from null type to expected type(s).